packages/ra-auth-google/Readme.md

```diff
@@ -0,0 +1,581 @@
+      loginPage={LoginPage}
+    >
+      <Resource name="posts" {...posts} />
+    </Admin>
+  );
+};
+export default App;
+```
+
+**Tip:** This example uses the [`<Login>`](https://marmelab.com/react-admin/Authentication.html#customizing-the-login-component) component provided by react-admin to create the login page. You can also include the `<LoginButton>` component in your own login page if you prefer.
+
+![LoginPage](img/LoginPage.png)
+
+## Advanced Usage
+
+### Provide the Client ID via a prop
+
+You can also provide the client id via a prop instead of using the `.env` file.
+
+```tsx
+// in src/App.tsx
+import React from 'react';
+import { Admin, Resource, Login } from 'react-admin';
+import { initGoogleAuthProvider } from "ra-auth-google";
+import dataProvider from './dataProvider';
+import posts from './posts';
+
+const App = () => {
+  const { authProvider, LoginButton } = initGoogleAuthProvider({
+    client_id: "my-application-client-id.apps.googleusercontent.com",
+  });
+
+  const LoginPage = () => (
+    <Login>
+      <LoginButton />
+    </Login>
+  );
+
+  return (
+    <Admin
+      authProvider={authProvider}
+      dataProvider={dataProvider}
+      title="Example Admin"
+      loginPage={LoginPage}
+    >
+      <Resource name="posts" {...posts} />
+    </Admin>
+   );
+};
+export default App;
+```
+
+### One Tap button
+
+The `<OneTapButton>` component can be used to enable the [One Tap](https://developers.google.com/identity/gsi/web/guides/offerings?hl=en#one_tap) button on one or more pages.
+
+![OneTapButton](img/GoogleOneTapButton.png)
+
+It can be used as a standalone component, or to wrap a page.
+
+```tsx
+const Standalone = () => (
+  <div>
+    <OneTapButton />
+    <h1>My Page</h1>
+  </div>
+);
+
+const Wrapper = () => (
+  <OneTapButton>
+    <div>
+      <h1>My Page</h1>
+    </div>
+  </OneTapButton>
+);
+```
+
+Here is a full example enabling the One Tap button on a custom route:
+
+```tsx
+// in src/App.tsx
+import React from "react";
+import { Admin, Resource, Login, CustomRoutes } from "react-admin";
+import { Route } from "react-router-dom";
+import { initGoogleAuthProvider } from "ra-auth-google";
+import dataProvider from "./dataProvider";
+import posts from "./posts";
+
+const App = () => {
+  const { authProvider, LoginButton, OneTapButton } = initGoogleAuthProvider();
+
+  const LoginPage = () => (
+    <Login>
+      <LoginButton />
+    </Login>
+  );
+
+  return (
+    <Admin
+      authProvider={authProvider}
+      dataProvider={dataProvider}
+      title="Example Admin"
+      loginPage={LoginPage}
+    >
+      <Resource name="posts" {...posts} />
+      <CustomRoutes>
+        <Route
+          path="/custom"
+          element={
+            <div>
+              <OneTapButton />
+              <h1>My Page</h1>
+            </div>
+          }
+        />
+      </CustomRoutes>
+    </Admin>
+  );
+};
+export default App;
+```
+
+### Customizing the Sign in with Google button
```

Comment:
```suggestion
### Customizing The Google Sign-in Button
```

Reply:
["Sign in with Google button"](https://developers.google.com/identity/gsi/web/guides/offerings?hl=en#sign_in_with_google_button) is the name under which this type of button is referenced in the Google docs. I think it's better to use the same vocabulary as them.

Reply:
Right! Then you just need to pass it to Title Case